fs/ext4/inode.c

```diff
@@ -3825,13 +3827,15 @@ static int __ext4_block_zero_page_range(handle_t *handle,
 
 	if (!buffer_uptodate(bh)) {
 		err = -EIO;
+		decrypt = S_ISREG(inode->i_mode) &&
+			ext4_encrypted_inode(inode);
+		ll_rw_block(REQ_OP_READ, (decrypt ? REQ_NOENCRYPT : 0), 1, &bh);
 		ll_rw_block(REQ_OP_READ, 0, 1, &bh);
 		wait_on_buffer(bh);
 		/* Uhhuh. Read error. Complain and punt. */
 		if (!buffer_uptodate(bh))
 			goto unlock;
-		if (S_ISREG(inode->i_mode) &&
-		    ext4_encrypted_inode(inode) &&
+		if ((decrypt) &&
```

Comment:
I don't see much of a reason to have parentheses around `decrypt`, especially since `&& !fscrypt_using_hardware_encryption(inode)` should probably be added to `decrypt`'s initialization like I mention above.

Reply:
It didn't build without the parentheses.

Reply:
Are you sure?

This change compiles fine for me with `aosp_yoshino_maple_defconfig`:

```diff
diff --git a/fs/ext4/inode.c b/fs/ext4/inode.c
index 7cf1a56c93be..e7f054639d28 100644
--- a/fs/ext4/inode.c
+++ b/fs/ext4/inode.c
@@ -3835,7 +3835,7 @@ static int __ext4_block_zero_page_range(handle_t *handle,
                /* Uhhuh. Read error. Complain and punt. */
                if (!buffer_uptodate(bh))
                        goto unlock;
-               if ((decrypt) &&
+               if (decrypt &&
                    !fscrypt_using_hardware_encryption(inode)) {
                        /* We expect the key to be set. */
                        BUG_ON(!fscrypt_has_encryption_key(inode));

```

Reply:
Oh, that's what you meant. Sorry I really misunderstood you there.

Reply:
Well, I now used your initialization method, so I'd say this is done.